src/ast/field_analyser.cpp

```diff
@@ -5,6 +5,16 @@
 namespace bpftrace {
 namespace ast {
 
+void FieldAnalyser::error(const std::string &msg, const location &loc)
+{
+  bpftrace_.error(err_, loc, msg);
```

Comment:
Why store in `err_` and not use `out_` like `warning()`?

Reply:
yep, should be out_, thanks

Reply:
@olsajiri 
I believe `err_` should be used here since FieldAnalyzer later checks if an error occurs by consulting `err_` length:
https://github.com/iovisor/bpftrace/blob/f0701114e22709ad4828c8c83f84f467e558e670/src/ast/field_analyser.cpp#L369
Now `semantic_analyser_btf.kfunc` test fails because field analyzer always returns zero.